packages/api/src/user/transformers.ts

```diff
@@ -9,40 +11,65 @@ import { generateUserName } from './utils/entity';
  * @param {UserRole[]} [currentUserRoles=[UserRole.USER]]
  * @returns {Promise<UserDto>}
  */
-export const userTransformer = async (
-  userDocument: UserDocument,
-  currentUserRoles: UserRole[] = [UserRole.USER]
-): Promise<UserDto> => {
-  const { _id, roles, createdAt, updatedAt } = userDocument;
-
-  /* Only return eth address to admin or quantifier */
-  let ethereumAddress;
-  if (
-    currentUserRoles.includes(UserRole.ADMIN) ||
-    currentUserRoles.includes(UserRole.QUANTIFIER)
-  ) {
-    ethereumAddress = userDocument.ethereumAddress;
-  }
+export const userTransformer = (userDocument: UserDocument): UserDto => {
+  const {
+    _id,
+    roles,
+    username,
+    createdAt,
+    updatedAt,
+    identityEthAddress,
+    rewardsEthAddress,
+  } = userDocument;
 
   let accounts;
   if (userDocument.accounts) {
     accounts = userAccountListTransformer(userDocument.accounts);
   }
 
-  // Generate user name
-  const nameRealized = await generateUserName(userDocument);
-
   return {
     _id,
     roles,
-    ethereumAddress,
+    identityEthAddress,
+    rewardsEthAddress,
     accounts,
-    nameRealized,
+    username,
     createdAt: createdAt.toISOString(),
     updatedAt: updatedAt.toISOString(),
   } as UserDto;
 };
 
+/**
+ * Serialize a User
+ *
+ * @param {UserDocument} userDocument
+ * @param {UserRole[]} [currentUserRoles=[UserRole.USER]]
+ * @returns {Promise<UserDto>}
+ */
+export const userDetailTransformer = async (
+  userDocument: UserDocument
+): Promise<UserDetailsDto> => {
+  const user = userTransformer(userDocument) as UserDetailsDto;
+  const account = user.accounts?.[0];
```

Comment:
Transformer needs to work if the User has more than one account - Discord and Telegram for instance.